runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -483,6 +494,28 @@ public void onChatMessage(ChatMessage event)
 		{
 			removeGameTimer(STAFF_OF_THE_DEAD);
 		}
+
+		if (config.showFreezes() && event.getMessage().equals(FROZEN_MESSAGE))
+		{
+			createGameTimer(ICEBARRAGE, false, true);
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		Player player = client.getLocalPlayer();
+		WorldPoint currentWorldPoint = player.getWorldLocation();
+
+		if (lastWorldX != currentWorldPoint.getX() || lastWorldY != currentWorldPoint.getY())
+		{
+			removeGameTimer(ICERUSH);
```

Comment:
Make a method to remove ancient freeze timers, and don't call `removeGameTimer` every tick unless it's needed. (Pollutes debug log pretty heavily)

Reply:
How do you feel about checking `infoBoxManager` for existing timers, this way we don't call `removeTimer` unless there is something to remove? 

We could also keep a struct of active timers within `TimersPlugin` although this would require some housekeeping.

These may be an overkill since we don't want to do too much on every game tick, but not sure.

Reply:
The first sentence is kind of what I was thinking--rather than directly calling `removeGameTimer` on each ice timer, check to see if they're active first, then remove them if they exist. The overhead should be fairly minimal since `removeGameTimer` would perform the same checks anyway.

Reply:
Ok cool, before pushing I tested the code and noticed a slight bug with this function. When running and frozen the timer doesn't show up until the next freeze cast (if at all). This could be an issue for hybrid fights especially. Not sure if this is because of WorldPoint (server/delayed) vs. LocalPoint (no delay). I'll try it out. 

We could also check a larger distance. If a player is immune for the remainder of the in game freeze timer (say if caster runs off screen) then we can entirely remove this function.

Reply:
Local point sees same behavior

Reply:
To my knowledge, that behavior is caused by PID. I believe that if you have it, you will be able to run for one additional tick before the freeze actually takes effect.